infrahub_sdk/schema/__init__.py

```diff
@@ -102,6 +110,14 @@ def validate_data_against_schema(self, schema: MainSchemaTypesAPI, data: dict) -
                     message=f"{key} is not a valid value for {identifier}",
                 )
 
+    def set_cache(self, schema: dict[str, Any] | BranchSchema, branch: str | None = None) -> None:
+        branch = branch or self.client.default_branch
+
+        if isinstance(schema, dict):
+            schema = BranchSchema.from_api_response(data=schema)
```

Comment:
In a way I'd prefer if .set_cache took a BranchSchema object directly. That way there's no confusion regarding what the `dict[str, Any]` should look like. It would mostly be relevant if there are any errors in the schema, then those errors would be raised in the correct location. On the other hand it might be slightly easier for some users if they don't have to convert it into a dict.

Reply:
`BranchSchema` is primarily an internal object so in most cases I would expect users to use the `dict` most of the time.
To clarify, I've added support for `SchemaRootAPI` as well which is the format returned by the `/api/schema` endpoint